[BUG] [go-server][golang] Trivial API file causes the generated code not to compile, due to missing imports "io" and "errors"

##### Description
The code generated doesn't compile, with the following error message:
```shell
go/api_silly_api.go:65:50: undefined: errors
go/api_silly_api.go:65:65: undefined: io
```

##### openapi-generator version
- 7.6.0
- 7.7.0
- 8.0.0-20240527.073301-5
They all exhibit the same behaviour.

##### OpenAPI declaration file content or url
https://gist.github.com/Andrei-Errapart/8a7eeaf6744aad7409a054df6f7fc398


##### Generation Details

All default options.

##### Steps to reproduce

```shell
rm -rf sillyserver
mkdir -p sillyserver
java -jar $jar generate -i api-1.0.1.yaml -g go-server --output sillyserver || exit $?
cd sillyserver
go get || exit $?
go build . || exit $?
```

##### Related issues/PRs
[#8847](https://github.com/OpenAPITools/openapi-generator/issues/8847)

Perhaps the same trick can be employed here: simply assign unused imports 
##### Suggest a fix
Here is one recommendation: [#8847 possible fix](https://github.com/OpenAPITools/openapi-generator/issues/8847#issuecomment-868158219)

The idea is to use to mark the imports as used and generate them always:
```go
// Reference imports to suppress errors if they are not otherwise used.
var _ = proto.Marshal
var _ = fmt.Errorf
var _ = math.Inf
```

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenOperation.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenOperation.java
@@ -116,6 +116,15 @@ public class CodegenOperation {
     }
 
     /**
+     * Check if there's at least one optional body parameter
+     *
+     * @return true if optional body parameter exists, false otherwise
+     */
+    public boolean getHasOptionalBodyParam() {
+        return nonEmpty(bodyParams) && nonEmpty(optionalParams) && bodyParams.stream().anyMatch(optionalParams::contains);
+    }
+
+    /**
      * Check if there's at least one query parameter
      *
      * @return true if query parameter exists, false otherwise

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoServerCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoServerCodegen.java
@@ -410,6 +410,7 @@ public class GoServerCodegen extends AbstractGoCodegen {
     private void addConditionalImportInformation(OperationsMap operations) {
         boolean hasPathParams = false;
         boolean hasBodyParams = false;
+        boolean hasOptionalBodyParams = false;
 
         for (CodegenOperation op : operations.getOperations().getOperation()) {
             if (op.getHasPathParams()) {
@@ -418,10 +419,14 @@ public class GoServerCodegen extends AbstractGoCodegen {
             if (op.getHasBodyParam()) {
                 hasBodyParams = true;
             }
+            if (op.getHasOptionalBodyParam()) {
+                hasOptionalBodyParams = true;
+            }
         }
 
         additionalProperties.put("hasPathParams", hasPathParams);
         additionalProperties.put("hasBodyParams", hasBodyParams);
+        additionalProperties.put("hasOptionalBodyParams", hasOptionalBodyParams);
     }
 
 

--- a/modules/openapi-generator/src/main/resources/go-server/controller-api.mustache
+++ b/modules/openapi-generator/src/main/resources/go-server/controller-api.mustache
@@ -6,12 +6,10 @@ import (
 	{{#hasBodyParams}}
 	"encoding/json"
 	{{/hasBodyParams}}
-	{{#isBodyParam}}
-	    {{^required}}
+	{{#hasOptionalBodyParams}}
 	"errors"
 	"io"
-	    {{/required}}
-	{{/isBodyParam}}
+	{{/hasOptionalBodyParams}}
 	"net/http"
 	"strings"
 {{#imports}}	"{{import}}"

